cmd/otel-allocator/prehook/relabel.go

```diff
@@ -82,6 +86,19 @@ func (tf *RelabelConfigTargetFilter) SetConfig(cfgs map[string][]*relabel.Config
 	for key, val := range cfgs {
 		relabelCfgCopy[key] = val
 	}
+
+	out, err := yaml2.Marshal(relabelCfgCopy)
+	if err != nil {
+		tf.log.V(2).Info("Error Marshaling", "error", err)
+		return
+	}
+
+	byteArr := replaceShard([]byte(out))
+	err = yaml2.Unmarshal(byteArr, &relabelCfgCopy)
+	if err != nil {
+		tf.log.Info("Error Unmarshalling", "error", err)
+		return
+	}
```

Comment:
I'm worried here regarding performance.    
Do you have investigated if it's possible to set the chart in the loop above?

Reply:
Thanks for the review! I took your suggestion and moved this logic inside the loop.